### PR TITLE
SUBMARINE-1395. Remove unauthenticated login

### PR DESCRIPTION
**Pull request by cdmikechen**
### What is this PR for?
Backend server no longer supports login without authentication (except python-sdk), REST must authenticate first (/api/auth/token) and add a token in header to be accessed.

### What type of PR is it?
Feature 

### Todos
* [x] - Add `/api/auth/token` endpoint and sdk codes
* [x] - Remove none auth type
* [x] - Fix test cases  
* [x] - Reformat openapi.json by order fileds and tags (correct ordering will prevent each gen-sdk openapi.json having a different order)

### What is the Jira issue?
https://issues.apache.org/jira/browse/SUBMARINE-1395

### How should this be tested?
CI test

### Screenshots (if appropriate)
NA

### Questions:
* Do the license files need updating? No
* Are there breaking changes for older versions? Yes
* Does this need new documentation? No

### CODE REVIEW

File: submarine-server/server-core/src/test/java/org/apache/submarine/server/AbstractSubmarineServerTest.java [hunk 1]
```diff
@@ -237,10 +263,12 @@ protected static PostMethod httpPost(String path, String request, String mediaTy
     PostMethod postMethod = new PostMethod(URL + path);
     postMethod.setRequestBody(request);
     postMethod.setRequestHeader("Content-type", mediaType);
-    postMethod.getParams().setCookiePolicy(CookiePolicy.IGNORE_COOKIES);
+    //postMethod.getParams().setCookiePolicy(CookiePolicy.IGNORE_COOKIES);
```
Comment (pingsutw) [inline submarine-server/server-core/src/test/java/org/apache/submarine/server/AbstractSubmarineServerTest.java:266]: Should we delete it?

Comment (cdmikechen) [reply submarine-server/server-core/src/test/java/org/apache/submarine/server/AbstractSubmarineServerTest.java:266]: > Should we delete it?

Of course, with REST we should no longer explicitly support session/cookie later on.

Comment (pingsutw) [reply submarine-server/server-core/src/test/java/org/apache/submarine/server/AbstractSubmarineServerTest.java:266]: SGTM

**Comment by cdmikechen** [comment]
It looks like we meet `no space left on device` error again. 
https://github.com/apache/submarine/actions/runs/5693116661/job/15444228405?pr=1092

image

**Review by pingsutw — APPROVED** [review]

**Comment by cdmikechen** [comment]
Currently the following error occurs inside the new ci testing process: 
https://github.com/apache/submarine/actions/runs/5840601129/job/15840001205
image

This is caused by some data within submarine_test that shouldn't be there. I'm currently looking for the exact cause.

**Comment by cdmikechen** [comment]
Since some of the `submarine-server-core` tests are started based on the built-in jetty, that test cases do not go through the hard-coded logic of `org.apache.submarine.server.database.utils.checkCalledByTestMethod` at startup, making the startup of the test case go to submarine's database, causing a database processing exception.

https://github.com/apache/submarine/blob/f170037c1e7642c16cd5fa1707e85abd6252d839/submarine-server/server-database/src/main/java/org/apache/submarine/server/database/utils/MyBatisUtil.java#L83-L104

**Comment by cdmikechen** [comment]
A related issue https://issues.apache.org/jira/browse/SUBMARINE-1297 has already been created before, and is currently being temporarily resolved by setting the environment variable for `JDBC_URL=jdbc:mysql://127.0.0.1:3306/submarine_test?useUnicode=true&characterEncoding=UTF-8&autoReconnect=true&allowMultiQueries=true&failOverReadOnly=false&zeroDateTimeBehavior=convertToNull&useSSL=false`. 
This should be fully resolved when the subsequent issue is resolved.

**Pull request closed without merging**